com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/IEvaluationProvider.java

```diff
@@ -41,6 +42,17 @@
      */
     CompletableFuture<Value> evaluate(String expression, ThreadReference thread, int depth);
 
+    /**
+     * Evaluate the expression at the given context of the object instance, return the promise which is to be resolved/rejected when
+     * the evaluation finishes.
+     * @param expression The expression to be evaluated
+     * @param context The jdi object which the expression will be executed at
+     * @param thread The jdi thread which the expression will be executed at
+     * @param depth The depth of stackframe of the stopped thread
+     * @return the evaluation result future
+     */
+    CompletableFuture<Value> evaluate(String expression, ObjectReference context, ThreadReference thread, int depth);
```

Comment:
rename this function to evaluateOnObject, two functions with same name and trivial differences in parameters is not good, caller will get confused.